src/main/resources/static/js/hposummary.js

```diff
@@ -69,5 +80,18 @@ $(document).ready(function() {
 			$(this).html('<i class="fas fa-minus-square"></i>');
 		}
 	});
-
+	
+	$('#hposummary tbody').on( 'click', 'button', function () {
+ 		var tr = $(this).closest('tr');
+		var row = table.row(tr);
+        var data = row.data();
+        console.log(data);
+        $('#CommentsModal').modal("show");
```

Comment:
Two observations: First, these element IDs are kind of generic. Perhaps they should match the modal name to better prevent name clashes, e.g. "#commentsmodal-title". Second, you might want to reset the form fields before showing the modal to ensure the user doesn't see a flash of the previous state. Probably not noticeable due to the modal's animation though.